test/unit/overlay.js

```diff
@@ -264,6 +264,146 @@ describe('Overlays', function() {
+        .overlayWith(fixtures.inputPngWithTransparency16bit, {
+          top: 1000
+        })
+        .toBuffer(function(err, data, info) {
+          if (err) throw err;
+          assert.strictEqual('jpeg', info.format);
+          assert.strictEqual(3, info.channels);
+          fixtures.assertSimilar(expected, data, done);
+        });
+      
+    });
+
+    it('Overlay with only left offset', function(done) {
+      var expected = fixtures.expected('overlay-only-left-offset.jpg');
+      sharp(fixtures.inputJpg)
+        .resize(400)
+        .overlayWith(fixtures.inputPngWithTransparency16bit, {
+          left: 1000
+        })
+        .toBuffer(function(err, data, info) {
+          if (err) throw err;
+          assert.strictEqual('jpeg', info.format);
+          assert.strictEqual(3, info.channels);
+          fixtures.assertSimilar(expected, data, done);
+        });
+      
+    });
+
+    it('Overlay with negative offsets', function(done) {
+      var expected = fixtures.expected('overlay-negative-offset.jpg');
+      sharp(fixtures.inputJpg)
+        .resize(400)
+        .overlayWith(fixtures.inputPngWithTransparency16bit, {
+          top: -1000,
+          left: -1000
+        })
+        .toBuffer(function(err, data, info) {
+          if (err) throw err;
+          assert.strictEqual('jpeg', info.format);
+          assert.strictEqual(3, info.channels);
+          fixtures.assertSimilar(expected, data, done);
+        });
+      
+    });
+
+    it('Overlay with 0 offset', function(done) {
+      var expected = fixtures.expected('overlay-offset-0.jpg');
+      sharp(fixtures.inputJpg)
+        .resize(400)
+        .overlayWith(fixtures.inputPngWithTransparency16bit, {
+          top: 0,
+          left: 0
+        })
+        .toBuffer(function(err, data, info) {
+          if (err) throw err;
+          assert.strictEqual('jpeg', info.format);
+          assert.strictEqual(3, info.channels);
+          fixtures.assertSimilar(expected, data, done);
+        });
+      
+    });
+
+    it('Overlay with offset and gravity', function(done) {
+      var expected = fixtures.expected('overlay-offset-with-gravity.jpg');
+      sharp(fixtures.inputJpg)
+        .resize(400)
+        .overlayWith(fixtures.inputPngWithTransparency16bit, {
+          left: 10,
+          top: 10,
+          gravity : 4
+          
+        })
+        .toBuffer(function(err, data, info) {
+          if (err) throw err;
+          assert.strictEqual('jpeg', info.format);
+          assert.strictEqual(3, info.channels);
+          fixtures.assertSimilar(expected, data, done);
+        });
+      
+    });
+
+    it('Overlay with offset and gravity and tile', function(done) {
+      var expected = fixtures.expected('overlay-offset-with-gravity-tile.jpg');
+      sharp(fixtures.inputJpg)
+        .resize(400)
+        .overlayWith(fixtures.inputPngWithTransparency16bit, {
+          left: 10,
+          top: 10,
+          gravity : 4,
```

Comment:
if both top/left and gravity are present and valid, top/left takes preference over gravity. Neither of the inputs individually are invalid. So IMO, the test shouldn't fail. But should work as per the set preference.